src/binary_heap.rs

```diff
@@ -1086,6 +1097,47 @@ impl<T, C: Compare<T>> BinaryHeap<T, C> {
 
         self.rebuild_tail(start);
     }
+
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all elements `e` for which `f(&e)` returns
+    /// `false`. The elements are visited in unsorted (and unspecified) order.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use std::collections::BinaryHeap;
+    ///
+    /// let mut heap = BinaryHeap::from([-10, -5, 1, 2, 4, 13]);
+    ///
+    /// heap.retain(|x| x % 2 == 0); // only keep even numbers
+    ///
+    /// assert_eq!(heap.into_sorted_vec(), [-10, 2, 4])
+    /// ```
+    // #[stable(feature = "binary_heap_retain", since = "1.70.0")]
```

Comment:
```suggestion
```

Reply:
This crate takes in another generic `C: Compare` since it supports different kinds of heap (instead of just max), like taking in a comparator function so the heap is sorted by the function. std on the other hand, contain another generic trait `A: Allocator` and use `T: Ord`.

Here's one of the example diff:

```diff
- impl<T: Ord, A: Allocator> Drop for RebuildOnDrop<'_, T, A> {
+ impl<T, C: Compare<T>> Drop for RebuildOnDrop<'_, T, C> {
```